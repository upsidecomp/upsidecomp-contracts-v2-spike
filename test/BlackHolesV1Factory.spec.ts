import { ethers, waffle } from 'hardhat';
import { BlackHoles } from '../typechain/BlackHoles';
import { expect } from './shared/expect';
import snapshotGasCost from './shared/snapshotGasCost';

import { FeeAmount, getCreate2Address } from './shared/utilities';

const { constants } = ethers;

const createFixtureLoader = waffle.createFixtureLoader;

const POOL_OWNER = '0x1000000000000000000000000000000000000000';

describe('BlackHolesV1Factory', () => {
  const [wallet, other] = waffle.provider.getWallets();

  let blackholes: BlackHoles;
  let poolBytecode: string;

  const fixture = async () => {
    const bh = await ethers.getContractFactory('BlackHoles');
    return (await bh.deploy()) as BlackHoles;
  };

  let loadFixture: ReturnType<typeof createFixtureLoader>;
  before('create fixture loader', async () => {
    loadFixture = createFixtureLoader([wallet, other]);
  });

  before('load pool bytecode', async () => {
    poolBytecode = (await ethers.getContractFactory('BlackHoles')).bytecode;
  });

  beforeEach('deploy factory', async () => {
    blackholes = await loadFixture(fixture);
  });

  it('owner is deployer', async () => {
    // expect(await blackholes.owner()).to.eq(wallet.address);
  });

  it('factory bytecode size', async () => {
    expect(((await waffle.provider.getCode(blackholes.address)).length - 2) / 2).to.matchSnapshot();
  });
});
