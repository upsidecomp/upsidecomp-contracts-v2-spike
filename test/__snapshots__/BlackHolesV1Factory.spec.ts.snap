// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlackHolesV1Factory factory bytecode size 1`] = `5053`;
